api/src/main/java/org/apache/iceberg/actions/RewriteDeletes.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.actions;
+
+import java.util.Set;
+import org.apache.iceberg.DeleteFile;
+
+public interface RewriteDeletes extends SnapshotUpdate<RewriteDeletes, RewriteDeletes.Result> {
+
+  /**
+   * Set the implementation class name for rewrite strategy.
+   *
+   * @return this for method chaining
+   */
+  RewriteDeletes strategy(String strategyImpl);
+
+  /**
+   * The action result that contains a summary of the execution.
+   */
+  interface Result {
+    /**
+     * Returns the delete files to rewrite.
+     */
+    Set<DeleteFile> deleteFilesToReplace();
```

Comment:
Based on the name of these methods, I assume the commit will happen outside of the action? I think this is different compared to what @RussellSpitzer has in the new data compaction. 

I believe the commit logic we use during the data compaction would also apply here as we can compact individual partitions concurrently. I'd love to see the same optimizations we added for data file rewrites. They make a big difference for large tables.

@RussellSpitzer, thoughts?

Reply:
it is committed in the action, see the action implementation here: https://github.com/apache/iceberg/pull/2364/files#diff-0f115e8104a36761f3afa09592e7beb70dcfeca2e5b088a05ed76a5778bf6178R84. 

The `option()` is used to extend features like that. We can implement partially committing feature in the specific strategy according to the options passed in. I have considered this before.